transport_analysis/analysis/velocityautocorr.py

```diff
@@ -5,111 +5,146 @@
-        AtomGroup
+    atomgroup : AtomGroup
+        An MDAnalysis :class:`~MDAnalysis.core.groups.AtomGroup`.
+        Note that :class:`UpdatingAtomGroup` instances are not accepted.
+    dim_type : {'xyz', 'xy', 'yz', 'xz', 'x', 'y', 'z'}
+        Desired dimensions to be included in the VACF. Defaults to 'xyz'.
+    fft : bool
+        If ``True``, uses a fast FFT based algorithm for computation of
+        the VACF. Otherwise, use the simple "windowed" algorithm.
+        The tidynamics package is required for `fft=True`.
+        Defaults to ``True``.
 
     Attributes
     ----------
-    universe: :class:`~MDAnalysis.core.universe.Universe`
-        The universe to which this analysis is applied
-    atomgroup: :class:`~MDAnalysis.core.groups.AtomGroup`
+    atomgroup : :class:`~MDAnalysis.core.groups.AtomGroup`
         The atoms to which this analysis is applied
-    results: :class:`~MDAnalysis.analysis.base.Results`
-        results of calculation are stored here, after calling
-        :meth:`VelocityAutocorr.run`
-    start: Optional[int]
-        The first frame of the trajectory used to compute the analysis
-    stop: Optional[int]
-        The frame to stop at for the analysis
-    step: Optional[int]
-        Number of frames to skip between each analyzed frame
-    n_frames: int
-        Number of frames analysed in the trajectory
-    times: numpy.ndarray
+    dim_fac : int
+        Dimensionality :math:`d` of the VACF.
+    results.timeseries : :class:`numpy.ndarray`
+        The averaged VACF over all the particles with respect to lag-time.
+        Obtained after calling :meth:`VelocityAutocorr.run`
+    results.vacf_by_particle : :class:`numpy.ndarray`
+        The VACF of each individual particle with respect to lag-time.
+    start : Optional[int]
+        The first frame of the trajectory used to compute the analysis.
+    stop : Optional[int]
+        The frame to stop at for the analysis.
+    step : Optional[int]
+        Number of frames to skip between each analyzed frame.
+    n_frames : int
+        Number of frames analysed in the trajectory.
+    n_particles : int
+        Number of particles VACF was calculated over.
+    times : numpy.ndarray
         array of Timestep times. Only exists after calling
         :meth:`VelocityAutocorr.run`
-    frames: numpy.ndarray
+    frames : numpy.ndarray
         array of Timestep frame indices. Only exists after calling
         :meth:`VelocityAutocorr.run`
     """
 
     def __init__(
         self,
-        universe_or_atomgroup: Union["Universe", "AtomGroup"],
-        select: str = "all",
-        # TODO: add your own parameters here
+        atomgroup: "AtomGroup",
+        dim_type: str = "xyz",
+        fft: bool = True,
         **kwargs
-    ):
+    ) -> None:
         # the below line must be kept to initialize the AnalysisBase class!
-        super().__init__(universe_or_atomgroup.trajectory, **kwargs)
```

Comment:
Lets consider making a decorator that returns `u.atoms` if the input is a universe, that way you can accept both.

Reply:
What do you mean by decorator? I was following the feedback in my proposal about avoiding having a `Universe` with a selection string and only using an `AtomGroup` instead.

Reply:
Just a thought I have been throwing around, would look like 

```python
@universe_to_atomgroup # decorator
def my_func_that_takes_u_or_ag(input):
      # decorator converts input universe to `atomgroup` if it isnt one already
      do_something_with_atomgroup(input)
```

But tbh this is just a random nice to have, and should probably go in MDA-core, so ignore me for now hahahah.

Reply:
If this were my code I'd only accept AtomGroup and not Universe. AtomGroup is significantly more flexible than Universe and feels conceptually like the right thing here as we are ultimately interested in the VACF of the particles. It's not much work to support Universe, but it's even less work to turn a universe into an atom group (`u.atoms`). Ultimately a personal preference.